File: py/routes/test_head_me.py
# -*- coding: utf-8 -*-

import pytest
from base_api_utils import BaseApiUtils


class TestBAUHeadMe:
    @pytest.fixture
    def bau(self, cfg):
        self.URL = cfg["base_url"] + '/head-me'
        self.DEFAULT_ERR_BODY = cfg["default_err_body"]
        self.EXPECTED_BODY = {"message": "headed"} 
        self.EXPECTED_ALLOW_VALUES = ['HEAD', 'OPTIONS']
        return BaseApiUtils(cfg)

    def test_head_me_delete(self, bau):
        rsp = bau.delete_it(self.URL)
        bool, msg = bau.check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = bau.check_response_body(rsp=rsp, expected_body=self.DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = bau.check_response_content_type(
            rsp=rsp,
            expected_header_value='application/json'
        )
        assert bool, msg
        for allow_val in self.EXPECTED_ALLOW_VALUES: 
            bool, msg = bau.check_response_allow_methods(
                rsp=rsp,
                expected_header_value=allow_val
            )
            assert bool, msg

    def test_head_me_get(self, bau):
        rsp = bau.get_it(self.URL)
        bool, msg = bau.check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = bau.check_response_body(rsp=rsp, expected_body=self.DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = bau.check_response_content_type(
            rsp=rsp,
            expected_header_value='application/json'
        )
        assert bool, msg
        for allow_val in self.EXPECTED_ALLOW_VALUES: 
            bool, msg = bau.check_response_allow_methods(
                rsp=rsp,
                expected_header_value=allow_val
            )
            assert bool, msg

    def test_head_me_head(self, bau):
        rsp = bau.head_it(self.URL)
        bool, msg = bau.check_response_status_code(rsp=rsp, expected_status_code=200)
        assert bool, msg
        bool, msg = bau.check_response_body(rsp=rsp, expected_body='')
        assert bool, msg
        bool, msg = bau.check_response_content_type(
            rsp=rsp,
            expected_header_value='application/json'
        )
        assert bool, msg

    def test_head_me_options(self, bau):
        rsp = bau.options_it(self.URL)
        bool, msg = bau.check_response_status_code(rsp=rsp, expected_status_code=200)
        assert bool, msg
        bool, msg = bau.check_response_body(rsp=rsp, expected_body='')
        assert bool, msg
        bool, msg = bau.check_response_content_type(
            rsp=rsp,
            expected_header_value='text/html'
        )
        assert bool, msg
        for allow_val in self.EXPECTED_ALLOW_VALUES: 
            bool, msg = bau.check_response_allow_methods(
                rsp=rsp,
                expected_header_value=allow_val
            )
            assert bool, msg

    def test_head_me_patch(self, bau):
        rsp = bau.patch_it(self.URL)
        bool, msg = bau.check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = bau.check_response_body(rsp=rsp, expected_body=self.DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = bau.check_response_content_type(
            rsp=rsp,
            expected_header_value='application/json'
        )
        assert bool, msg
        for allow_val in self.EXPECTED_ALLOW_VALUES: 
            bool, msg = bau.check_response_allow_methods(
                rsp=rsp,
                expected_header_value=allow_val
            )
            assert bool, msg

    def test_head_me_post(self, bau):
        rsp = bau.post_it(self.URL)
        bool, msg = bau.check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = bau.check_response_body(rsp=rsp, expected_body=self.DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = bau.check_response_content_type(
            rsp=rsp,
            expected_header_value='application/json'
        )
        assert bool, msg
        for allow_val in self.EXPECTED_ALLOW_VALUES: 
            bool, msg = bau.check_response_allow_methods(
                rsp=rsp,
                expected_header_value=allow_val
            )
            assert bool, msg

    def test_head_me_put(self, bau):
        rsp = bau.put_it(self.URL)
        bool, msg = bau.check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = bau.check_response_body(rsp=rsp, expected_body=self.DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = bau.check_response_content_type(
            rsp=rsp,
            expected_header_value='application/json'
        )
        assert bool, msg
        for allow_val in self.EXPECTED_ALLOW_VALUES: 
            bool, msg = bau.check_response_allow_methods(
                rsp=rsp,
                expected_header_value=allow_val
            )
            assert bool, msg

File: rb/routes/test_head_me.rb
# encoding: utf-8

require_relative '../bau_init.rb'

class TS_BAU_HEAD_ME < ParallelTestCases
    def setup
        @URL = "#{ENV['BAU_BASE_URL']}/head-me"
        @DEFAULT_ERR_BODY = JSON.parse(ENV['DEFAULT_ERR_BODY']).to_json
        @EXPECTED_BODY = '{"message":"headed"}'
        @EXPECTED_ALLOW_VALUES = ['HEAD', 'OPTIONS']
    end

    def teardown
        # do teardown stuff
    end

    def test_head_me_delete
        rsp = BaseApiUtils.delete_it(@URL)  
        bool, msg = check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = check_response_body(rsp=rsp, expected_body=@DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = check_response_content_type(rsp=rsp, expected_header_value='application/json')
        assert bool, msg
        @EXPECTED_ALLOW_VALUES.each do | allow_val |
            bool, msg = check_response_allow_methods(rsp=rsp, expected_header_value=allow_val)
            assert bool, msg
        end
    end

    def test_head_me_get
        rsp = BaseApiUtils.get_it(@URL)  
        bool, msg = check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = check_response_body(rsp=rsp, expected_body=@DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = check_response_content_type(rsp=rsp, expected_header_value='application/json')
        assert bool, msg
        @EXPECTED_ALLOW_VALUES.each do | allow_val |
            bool, msg = check_response_allow_methods(rsp=rsp, expected_header_value=allow_val)
            assert bool, msg
        end
    end

    def test_head_me_head
        rsp = BaseApiUtils.head_it(@URL)  
        bool, msg = check_response_status_code(rsp=rsp, expected_status_code=200)
        assert bool, msg
        bool, msg = check_response_body(rsp=rsp, expected_body='')
        assert bool, msg
        bool, msg = check_response_content_type(rsp=rsp, expected_header_value='application/json')
        assert bool, msg
    end

    def test_head_me_options
        rsp = BaseApiUtils.options_it(@URL)  
        bool, msg = check_response_status_code(rsp=rsp, expected_status_code=200)
        assert bool, msg
        bool, msg = check_response_body(rsp=rsp, expected_body='')
        assert bool, msg
        bool, msg = check_response_content_type(rsp=rsp, expected_header_value='text/html')
        assert bool, msg
        @EXPECTED_ALLOW_VALUES.each do | allow_val |
            bool, msg = check_response_allow_methods(rsp=rsp, expected_header_value=allow_val)
            assert bool, msg
        end
    end

    def test_head_me_patch
        rsp = BaseApiUtils.patch_it(@URL)  
        bool, msg = check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = check_response_body(rsp=rsp, expected_body=@DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = check_response_content_type(rsp=rsp, expected_header_value='application/json')
        assert bool, msg
        @EXPECTED_ALLOW_VALUES.each do | allow_val |
            bool, msg = check_response_allow_methods(rsp=rsp, expected_header_value=allow_val)
            assert bool, msg
        end
    end

    def test_head_me_post
        rsp = BaseApiUtils.post_it(@URL)  
        bool, msg = check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = check_response_body(rsp=rsp, expected_body=@DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = check_response_content_type(rsp=rsp, expected_header_value='application/json')
        assert bool, msg
        @EXPECTED_ALLOW_VALUES.each do | allow_val |
            bool, msg = check_response_allow_methods(rsp=rsp, expected_header_value=allow_val)
            assert bool, msg
        end
    end

    def test_head_me_put
        rsp = BaseApiUtils.put_it(@URL)  
        bool, msg = check_response_status_code(rsp=rsp, expected_status_code=405)
        assert bool, msg
        bool, msg = check_response_body(rsp=rsp, expected_body=@DEFAULT_ERR_BODY)
        assert bool, msg
        bool, msg = check_response_content_type(rsp=rsp, expected_header_value='application/json')
        assert bool, msg
        @EXPECTED_ALLOW_VALUES.each do | allow_val |
            bool, msg = check_response_allow_methods(rsp=rsp, expected_header_value=allow_val)
            assert bool, msg
        end
    end

end # class